File: src/main/java/Main/ItemClasses/Weapons/Weapon.java
package Main.ItemClasses.Weapons;

import Main.ItemClasses.Item;

public class Weapon extends Item {
    private String damage;

    public Weapon() {
        super();
    }

    public String getDamage() {
        return damage;
    }

    public void setDamage(String damage) {
        this.damage = damage;
    }

    public Weapon(String name, String description, int weight, int valueInGold, int amount, String damage) {
        super(name, description, weight, valueInGold, amount);
        this.damage = damage;
    }
}

File: src/main/kotlin/main/ItemClasses/Weapons/Weapon.kt
package main.ItemClasses.Weapons

import main.ItemClasses.Item
import kotlinx.serialization.Serializable

@Serializable
open class Weapon() : Item() {
    var damage: String = ""

    constructor(name: String, description: String, weight: Int, valueInGold: Int, amount: Int, damage: String) : this() {
        this.name = name
        this.description = description
        this.weight = weight
        this.valueInGold = valueInGold
        this.amount = amount
        this.damage = damage
    }
}
